packages/block-library/src/heading/test/edit.native.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+
+/**
+ * WordPress dependencies
+ */
+import * as data from '@wordpress/data';
+import * as blockLibrary from '@wordpress/block-library';
+import * as blocks from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import HeadingToolbar from '../heading-toolbar';
+import HeadingEdit from '../edit';
+
+const blockEditorSelect = data.select( 'core/block-editor' );
+const blockEditorDispatch = data.dispatch( 'core/block-editor' );
+
+const testAttributes = {
+	content: 'Heading block',
+	level: 4,
+	align: 'center',
+	style: { color: { text: '#003fa3' } },
+};
+
+describe( 'Heading', () => {
+	beforeAll( () => {
+		blockLibrary.registerCoreBlocks();
+		gutenbergSetup();
+		blockEditorDispatch.removeBlocks( blockEditorSelect.getBlocks() );
+	} );
+
+	it( 'Serializes all Heading attributes properly', () => {
+		const expectedHTML = `<!-- wp:heading {"align":"center","level":4,"style":{"color":{"text":"#003fa3"}}} -->
+<h4 class="has-text-align-center has-text-color" style="--wp--color--text:#003fa3">Heading block</h4>
+<!-- /wp:heading -->`;
+
+		insertHeadingBlockWithAttributes( testAttributes );
+
+		const finalHTML = blocks.serialize( blockEditorSelect.getBlocks() );
+
+		expect( finalHTML ).toEqual( expect.stringContaining( expectedHTML ) );
+	} );
+
+	it( 'Parse Heading attributes from HTML properly', () => {
+		const expectedHTML = `<!-- wp:heading {"align":"center","level":4,"style":{"color":{"text":"#003fa3"}}} -->
+<h4 class="has-text-align-center has-text-color" style="--wp--color--text:#003fa3">Heading block</h4>
+<!-- /wp:heading -->`;
+
+		const parsedBlocks = blocks.parse( expectedHTML );
+
+		expect( parsedBlocks[ 0 ].attributes ).toEqual( testAttributes );
```

Comment:
Interesting test as well. I think I don't see the web codepaths to test the attributes that result from parsing block content, which makes me wonder again, why not 🤔.

Reply:
This test in particular is the kind of I'm less convinced about. I added it mostly for completeness sake. It might be good if it can ward us from possible breakages on mobile, but not sure how those breakages might look like.

Just as a reminder that we are running native code paths. If there is (or might be in the future) any web/native platform divergence in the path of this task, this is warding us from posible breakages.